entity-service-api/src/main/proto/org/hypertrace/entity/query/service/v1/entity_query_request.proto

```diff
@@ -69,3 +69,15 @@ message BulkEntityArrayAttributeUpdateRequest {
 
 message BulkEntityArrayAttributeUpdateResponse {
 }
+
+message DeleteEntitiesRequest {
```

Comment:
Entity query service should be read-only (that's the "query" part 😉 ) . Please use or extend if necessary the EDS APIs.

Reply:
hmmm interesting, I was told the difference is based on treating it as generic objects or primary object. Also entity query service already has update, bulk update apis.

Reply:
both services are generic. The update APIs in EQS are a mistake (for some reason we didn't want to call EDS from gateway service, so we duplicated the APIs there).

Reply:
> The update APIs in EQS are a mistake (for some reason we didn't want to call EDS from gateway service, so we duplicated the APIs there).

The update APIs in EQS also provides a way to directly update a specific attribute, and hence directly updating the specific field in mongo, instead of getting the entire object, and then updating the entire object, leading to memory pressure

EDS in my opinion always works with the entire entity object

Reply:
I rather treat EQS as entity service, which works generically for attributes, and not only being used for querying, but can be used to update/delete attributes

Still, not convinced, why we shouldn't have update/delete APIs in EQS

Reply:
> I could buy the argument that there shouldn't be a separate EDS and EQS just two different query APIs (one for entity objects, one for "rows" that supports aggregations)

Does it really matter then, where we keep the APIs? If the API design and implementation fits well with the EQS APIs, why not keep the APIs there? (for implementation simplicity)

For example: In case of delete API, the API could have been simple, by just working on the attributes in EQS, rather than having a one-of in EDS, because of the complex "id" transformation handling needed in EDS